From 380f268bad158c59cda2289e20dbe602d8cdf383 Mon Sep 17 00:00:00 2001
From: Petr Kubiznak <kubiznak.petr@elnico.cz>
Date: Thu, 10 Dec 2015 11:48:59 +0100
Subject: [PATCH 1/2] sdhc: Use 22K pullup for the Card Detect pin for higher
 reliability.

It lowers the noise which might potentially cause card detaching.
---
 arch/arm/plat-mxc/include/mach/iomux-mvf.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/arch/arm/plat-mxc/include/mach/iomux-mvf.h b/arch/arm/plat-mxc/include/mach/iomux-mvf.h
index cf64267..3d4c9a1 100644
--- a/arch/arm/plat-mxc/include/mach/iomux-mvf.h
+++ b/arch/arm/plat-mxc/include/mach/iomux-mvf.h
@@ -72,6 +72,8 @@ typedef enum iomux_config {
 
 #define MVF600_GPIO_GENERAL_CTRL (PAD_CTL_PKE | PAD_CTL_PUE | PAD_CTL_SPEED_MED | PAD_CTL_PUS_47K_UP | \
 		PAD_CTL_DSE_25ohm)
+#define MVF600_GPIO_GENERAL_CTRL_22k (PAD_CTL_PKE | PAD_CTL_PUE | PAD_CTL_SPEED_MED | PAD_CTL_PUS_22K_UP | \
+		PAD_CTL_DSE_25ohm)
 
 #define MVF600_FTM0_CH_CTRL	(PAD_CTL_SPEED_LOW | PAD_CTL_OBE_ENABLE | \
 		PAD_CTL_ODE | PAD_CTL_DSE_25ohm)
@@ -93,7 +95,7 @@ typedef enum iomux_config {
 /*set PTA7 as GPIO for sdhc card detecting*/
 #define MVF600_PAD134_PTA7__SDHC1_SW_CD			\
 		IOMUX_PAD(0x0218, 0x0218, 0, 0x0000, 0, \
-				MVF600_GPIO_GENERAL_CTRL | PAD_CTL_IBE_ENABLE)
+				MVF600_GPIO_GENERAL_CTRL_22k | PAD_CTL_IBE_ENABLE)
 
 /*I2C0*/
 #define MVF600_PAD36_PTB14__I2C0_SCL				\
-- 
2.1.0

